graphite/graphite.go

```diff
@@ -79,6 +82,13 @@ func (f *GraphitePublisher) Publish(metrics []plugin.Metric, cfg plugin.Config)
 		}
 		data := fmt.Sprintf("%v", m.Data)
 		logger.Debug("Metric ready to send %s:%s", key, data)
+
+		if strings.ContainsAny(key, illegal) {
+			log.Info("Original string is %s", key)
+			key = strings.Replace(key, illegal, "_", len(key))
```

Comment:
I see your point. For this to work, the illegal characters need to occur together. Which cannot be assumed. I like your idea of replacing each individual illegal char to a different legal character.